packages/react-devtools-shared/src/backend/fiber/renderer.js

```diff
@@ -1950,13 +1959,23 @@ export function attach(
 
     const indices = [];
     let index = 0;
+
     while (next !== null) {
       if (didStatefulHookChange(prev, next)) {
         indices.push(index);
       }
+
+      // useSyncExternalStore creates 2 internal hooks, but we only count it as 1 user-facing hook
+      if (isUseSyncExternalStoreHook(next)) {
+        if (next.next !== null) {
+          next = next.next;
+          prev = prev.next;
+        }
```

Comment:
We probably don't want to check on `next.next` presence here, it should be implied if the hook object is actually an internal representation of `useSyncExternalStore()`.

I would prefer it to crash instead, if this ever happens, so we spot the gap in our hook parsing logic first.